nautobot/docs/user-guide/administration/upgrading/from-v1/ipam-upgrade-guide.md

```diff
@@ -0,0 +1,191 @@
+
+### VRF is no longer used for uniqueness
+
+- The `rd` field must unique to each Namespace
+- Relationship from Prefix inverted; now many Prefixes can be assigned to one or more VRF in the same Namespace
+- One ore mor Devices now be assigned to a VRF
+    - A VRF must be assigned to a Device before it may be assigned to an Interface/VMInterface
+
+### IPAddress to Interface relationship was inverted
+
+> This also applies to `VMInterface` objects
+
+In Nautobot 1.x the relationship from an IPAddress to an Interface was done by way of a foreign key to `Interface` on the `IPAddress` object. This implementation was flawed in that if a need arose to assign the same IP address host value to multiple Interfaces, it required the creation of duplicate `IPAddress` objects with the same `host` address in order to assign each one to a different `Interface`.
+
+As of Nautobot 2.0, this relationships was inverted. Now an `Interface` has a many-to-many relationship to `IPAddresses`. This allows the same `IPAddress` object to be assigned to multiple `Interface` objects without the need to create duplicate `IPAdress` objects.
+
+### VRF is now assigned to Interface/VMInterface, not IPAddress
+
+- A new foreign key to `VRF` has been introduced to `Interface/VMInterface`
+    - A`VRF` must be assigned to a `Device` before it may be assigned to an `Interface`
+    - A`VRF` must be assigned to a `VirtualMachine` before it may be assigned to an `VMInterface`
+- This addresses a fundamental flaw in which an Interface could have multiple IPAddresses assigned with conflicting VRFs, which is impossible in practice.
+
+### Primary IPv4/IPv6 no longer unique
+
+- [#3939](https://github.com/nautobot/nautobot/issues/3939) - Changed `Device.primary_ip4` and `primary_ip6` fields from `OneToOneField`to `ForeignKey`, relaxing the uniqueness constraint.
+- [#3939](https://github.com/nautobot/nautobot/issues/3939) - Changed `VirtualMachine.primary_ip4` and `primary_ip6` fields from `OneToOneField` to `ForeignKey`, relaxing the uniqueness constraint.
+- Because now same IP used in multiple places requires this.
+
+## Preparing your IPAM Data for Nautobot 2.0
+
+- IPaddress VRF assignments must be unique
+- Parents and IPAddresses must share the same Namespace
+- IPAddress objects can no longer be orphaned
+    - The database migration will automatically create a parent prefix for IPAddresses that do not have an eligible parent. For example 1.2.3.4/32 will have a parent Prefix created of the same network and prefix_length e.g. 1.2.3.4/32.
+    - If you do not wish for these single host prefixes to be created, create a parent prefix of your desired size to contain them before upgrading
+- You cannot migrate Interfaces or VMInterfaces that have IPs with differing VRFs
+    - Make sure for interfaces with multiple IPs that each IP assigned to the same Interface share the same VRF or are not assigned a VRF
```

Comment:
I know this is WIP but as a suggestion it might be good to add "before/bad" and "after/corrected" examples for each of these?